homeassistant/components/notify/__init__.py

```diff
@@ -275,6 +276,13 @@ async def async_setup_platform(
         return True
 
     setup_tasks = [
+        async_setup_platform(
+            persistent_notification.DOMAIN,
+            {CONF_NAME: persistent_notification.NOTIFY_SERVICE_NAME},
+        )
+    ]
+
```

Comment:
That is still not possible. We have strict interface concept which avoid to do that. Solution will be to define this as after dependence over meta data. They is also load with default config.

Reply:
I am not entirely sure what you mean. If you are talking about `after_dependencies`, that is pointless as:
1. `persistent_notification` is in the whitelist of imports allowed without specifying as a dependency. Further more, only a single integration lists it as a dependency, despite multiple others using it in the setup method, in a way that requires it to have already been fully loaded. This is because:
2. `persistent_notification` is always loaded as the second loaded integration, immediately after `homeassistant`. This is done without regard to dependencies, and without regard to the config. 

In case your your objection was to something else, I've pushed up an alternative, that goes back to using `discovery.async_load_platform` approach from `persistent_notification`, using configuration on the `persistent_notification` to allow the user to manually avoid any naming conflicts.

Reply:
If however the above did answer your concerns, I'd be happy to revert to that commit again.